pennylane/labs/resource_estimation/templates/subroutines.py

```diff
@@ -48,25 +66,74 @@ def _resource_decomp(num_wires, **kwargs) -> Dict[CompressedResourceOp, int]:
 
     @property
     def resource_params(self) -> dict:
+        r"""Returns a dictionary containing the minimal information needed to compute the resources.
+
+        Resource parameters:
+            num_wires (int): the number of qubits the operation acts upon
+
+        Returns:
+            dict: dictionary containing the resource parameters
+        """
         return {"num_wires": len(self.wires)}
 
     @classmethod
     def resource_rep(cls, num_wires) -> CompressedResourceOp:
+        r"""Returns a compressed representation containing only the parameters of
+        the Operator that are needed to compute a resource estimation.
+
+        Args:
+            num_wires (int): the number of qubits the operation acts upon
+
+        Returns:
+            CompressedResourceOp: the operator in a compressed representation
+        """
         params = {"num_wires": num_wires}
         return CompressedResourceOp(cls, params)
 
     @staticmethod
     def tracking_name(num_wires) -> str:
+        r"""Returns the tracking name built with the operator's parameters."""
         return f"QFT({num_wires})"
 
 
 class ResourceQuantumPhaseEstimation(qml.QuantumPhaseEstimation, ResourceOperator):
-    """Resource class for QPE"""
+    r"""Resource class for QuantumPhaseEstimation (QPE).
+
+    Args:
+        base_class (ResourceOperator): The type of the operation corresponding to the
+            phase estimation unitary.
+        base_params (dict): A dictionary of parameters required to obtain the resources for
+            the phase estimation unitary.
+        num_estimation_wires (int): the number of wires used for measuring out the phase
+
+    Resources:
+        The resources are obtained from the standard decomposition of QPE as presented
+        in (section 5.2) `Nielsen, M.A. and Chuang, I.L. (2011) Quantum Computation and Quantum
+        Information <https://www.cambridge.org/highereducation/books/quantum-computation-and-quantum-information/01E10196D0A682A6AEFFEA52D53BE9AE#overview>`_.
+
+    .. seealso:: :class:`~.QuantumPhaseEstimation`
+
+    """
 
     @staticmethod
     def _resource_decomp(
         base_class, base_params, num_estimation_wires, **kwargs
     ) -> Dict[CompressedResourceOp, int]:
+        r"""Returns a dictionary representing the resources of the operator. The
+        keys are the operators and the associated values are the counts.
+
+        Args:
+            base_class (Type(ResourceOperator)): The type of the operation corresponding to the
```

Comment:
```suggestion
            base_class (ResourceOperator): The type of the operation corresponding to the
```

Do we need type here?

Reply:
Yes! We are passing the reference to the class, not an instance of the class.